includes/import-functions.php

```diff
@@ -132,58 +131,72 @@ function edd_import_coupon_csv_file() {
 						$validation_error = true;
 						$validation_msg  .= __('Discount Name is blank.', 'edd') . "; ";
 					}
-					
+
 					// discount code  blank
 					if ( 0 == strlen( $edd_discount_code ) ) {
 						$validation_error = true;
 						$validation_msg  .= __('Discount code is blank.', 'edd') . "; ";
 					}
-					
+
+					// Verify only accepted characters.
+					$sanitized_discount_code = preg_replace( '/[^a-zA-Z0-9-_]+/', '', $edd_discount_code );
+					if ( strtoupper( $edd_discount_code ) !== strtoupper( $sanitized_discount_code ) ) {
+						$validation_error = true;
+						$validation_msg  .= __('Discount code is invalid.', 'edd') . "; ";
+					}
+
 					// discount type is blank
 					if ( 0 == strlen( $edd_discount_type ) ) {
 						$validation_error = true;
 						$validation_msg  .= __('Discount type is blank.', 'edd') . "; ";
 					}
-					
-					// discount type is invaid
-					if ( ( $edd_discount_type != 'flat' ) && ( $edd_discount_type != 'percentage' ) ) {
+
+					// discount type is invalid
+					$discount_types = array( 'flat', 'percentage' );
+					if ( ! in_array( $edd_discount_type, $discount_types ) ) {
```

Comment:
```suggestion
					if ( ! in_array( $edd_discount_type, $discount_types, true ) ) {
```
I think it would be okay to use a strict check here.